### PR TITLE
Include sast check for dockerfiles

### PR DESCRIPTION
**Pull request by Tansito**
### Summary

This PR includes checks that validates our Dockerfiles configurations. 

### Details and comments
- Added a script that looks for vulnerabilities
- Include `conftest`

ref #690

### CODE REVIEW

**Review by psschwei — COMMENTED** [review]
This test isn't showing up for this PR for some reason...

image

**Comment by Tansito** [comment]
I think is because of this commit @psschwei : [1557de7](https://github.com/Qiskit-Extensions/quantum-serverless/pull/696/commits/1557de78aa84b4e50f70568fca9be3deff2bc445) (I needed to enable it to make it work). But you can check previous tests before that commit [here](https://github.com/Qiskit-Extensions/quantum-serverless/actions/runs/5332823203)

File: .github/scripts/dockerfile-security.rego
Comment (akihikokuroda) [inline .github/scripts/dockerfile-security.rego:1]: I'm a little curious.  Are these checks recommended ones?

Comment (Tansito) [reply .github/scripts/dockerfile-security.rego:1]: Yes, from the security team. I can share with you internally the link if you are interested, Aki.

**Comment by psschwei** [comment]
Just trying to think through the workflow here... if this test is only run on push to main, does that mean the flow would be something like this:

- PR opened
- other tests run and pass
- approved by reviewer
- merge button clicked
- docker verify test runs
- if docker verify passes, then merge

If that's right, if the docker verify test _fails_, then the PR author would have to make changes but the PR would still be approved... if the changes needed to pass a security review were non-trivial, I could see wanting another review before merging...

File: .github/scripts/dockerfile-security.rego
Comment (Tansito) [reply .github/scripts/dockerfile-security.rego:1]: Np, they seem relatively new. I didn't apply them before until start some weeks ago with the deployment.

**Comment by Tansito** [comment]
Now @psschwei you should be able to see the checks with the last change that I did, for example.

**Review by IceKhan13 — APPROVED** [review]

File: .github/workflows/docker-verify.yaml
Comment (Tansito) [reply .github/workflows/docker-verify.yaml:51]: nope with the current configuration, at least

Comment (Tansito) [reply .github/workflows/docker-verify.yaml:51]: I assume I can add a path for this file too, I will check 👍

Comment (Tansito) [reply .github/workflows/docker-verify.yaml:52]: Solved in [3637475](https://github.com/Qiskit-Extensions/quantum-serverless/pull/696/commits/3637475259eb16acd9132ee0c8d26369e895773e)

**Comment by Tansito** [comment]
I don't know why this error didn't appear in #699 but I will take a look now 😂

**Comment by Tansito** [comment]
Seems a bit random this error 🤷‍♂️

**Pull request merged**
